compiler/src/org.graalvm.compiler.jtt/src/org/graalvm/compiler/jtt/optimize/Switch03.java

```diff
@@ -0,0 +1,635 @@
+/*
+ * Copyright (c) 2009, 2018, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.graalvm.compiler.jtt.optimize;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.graalvm.compiler.jtt.JTTTest;
+import org.graalvm.compiler.lir.hashing.HashFunction;
+import org.graalvm.compiler.lir.hashing.Hasher;
+import org.junit.Test;
+
+import jdk.vm.ci.meta.JavaConstant;
+
+/*
+ * Tests optimization of hash table switches.
+ * Code generated by `Switch03.TestGenerator.main`
+ */
+public class Switch03 extends JTTTest {
```

Comment:
Please use a more descriptive name since this is specifically about the hash switch, like HashSwitchTest.  Also is there any reason this is a JTTest?  These are special variants of regular compiler tests that are exercised with their parameters bounds as constants in addition to a normal compile to test high level transforms.  I don't think you care about that here.  I think this should probably be over in org.graalvm.compiler.core.test.  Or if you wanted it to be a more specific test of the LIR you could make it a LIRTest and check for the existence of a HashTableSwitchOp in the final LIR.  Anyway, renaming is enough for now.  You can make any improvements in a follow on PR.